.circleci/config.yml

```diff
@@ -0,0 +1,151 @@
+version: 2.1
+
+orbs:
+  aws-s3: circleci/aws-s3@1.0.11
+
+only_branches: &only_branches
+  filters:
+    branches:
+      ignore: master
+
+only_master: &only_master
+  filters:
+    branches:
+      only: master
+
+only_v_tags: &only_v_tags
+  filters:
+    tags:
+      only: /^v.*/
+    branches:
+      ignore: /.*/
+
+mm_binary: &mm_binary
+  mm-binary: https://releases.mattermost.com/mattermost-platform/master/mattermost-enterprise-linux-amd64.tar.gz
+
+jobs:
+  build-ltops-and-test:
+    docker:
+      - image: circleci/golang:1.12
+    steps:
+      - checkout
+      - restore_cache:
+          key: dependencies-{{ checksum "go.sum" }}
+      - run:
+          name: Install dependencies
+          command: |
+            export PATH=$PATH:$GOPATH/bin
+            mkdir -p $GOPATH/bin
+            make install
+      - save_cache:
+          key: dependencies-{{ checksum "go.sum" }}
+          paths:
+            - $GOPATH/pkg/mod
+      - run:
+          name: Build package
+          command: make package
+      - run:
+          name: Test
+          command: |
+            set +x
+            ltops --help
+      - persist_to_workspace:
+          root: .
+          paths:
+            - dist
+
+  deploy:
+    docker:
+      - image: circleci/python:3.6
+    parameters:
+      s3-sync-to:
+        type: string
+    steps:
+      - attach_workspace:
+          at: .
+      - aws-s3/sync:
+          from: mattermost-load-test/dist/mattermost-load-test.tar.gz
+          to: << parameters.s3-sync-to >>
+          arguments: --cache-control no-cache
+
+  build-docker:
+    docker:
+      - image: circleci/buildpack-deps:stretch
+    parameters:
+      mm-binary:
+        type: string
+      loadtest-binary:
+        type: string
+      docker-tag:
+        type: string
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          command: >
+            docker build
+            --build-arg MM_BINARY=<< parameters.mm-binary >>
+            --build-arg LOADTEST_BINARY=<< parameters.loadtest-binary >>
+            -t mattermost/mattermost-loadtest:<< parameters.docker-tag >>
+            .
+            --no-cache
+      - run:
+          command: |
+            docker login --username ${DOCKER_USER} --password ${DOCKER_PASS}
+            docker push mattermost/mattermost-loadtest:<< parameters.docker-tag >>
+            docker logout
+            docker rmi -f mattermost/mattermost-loadtest:<< parameters.docker-tag >>
+
+workflows:
+  version: 2
+  build_and_deploy:
+    jobs:
+      - build-ltops-and-test
+
+      - deploy:
```

Comment:
deploy is not a good name, maybe `deploy-s3` or `upload-s3`

Reply:
true, fixed that.